aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -209,7 +209,61 @@ void ne_kernel(TensorIterator& iter) {
        });
     });
   }
+}
+
+void max2_kernel(TensorIterator& iter) {
+  if (iter.dtype() == ScalarType::Bool) {
+    cpu_kernel(iter,
+      [](bool a, bool b) -> bool {
+        return a || b;
+      });
+  } else if (isIntegralType(iter.dtype(), /*includeBool*/ false)) {
+    AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "max2_cpu", [&]() {
+      cpu_kernel_vec(iter,
+        [](scalar_t a, scalar_t b) -> scalar_t { return std::max(a, b); },
+        [](Vec256<scalar_t> a, Vec256<scalar_t> b) { return at::vec256::maximum(a, b); });
```

Comment:
I think you want to use Vec256<scalar_t>:: to call typed implementation instead of map based.

Reply:
The definition of `maximum` is actually under `at::vec256`. I agree this is a bit unclean; I can make a follow-up PR to make it a class member. https://github.com/pytorch/pytorch/blob/7e95b899801bf27409b77107aee41856eab3fee9/aten/src/ATen/cpu/vec256/vec256_double.h#L238